spec/unit/pool_manager_spec.rb

```diff
@@ -1668,6 +1668,76 @@
     end
   end
 
+  describe '#get_domain_for_pool' do
+    let(:provider_name) { 'mock_provider' }
+    context 'No domain is set' do
+      let(:config) { YAML.load(<<~EOT
+        ---
+        :config:
+        :providers:
+          :mock_provider:
+        :pools:
+          - name: '#{pool}'
+            size: 1
+            provider: #{provider_name}
+        EOT
+      )}
+
+      it 'should return nil' do
+        allow(subject).to receive(:get_provider_for_pool).with(pool).and_return(provider)
+        allow(provider).to receive(:name).and_return(provider_name)
+        result = subject.get_domain_for_pool(pool)
+        expect(result).to be_nil
+      end
+    end
+
+    context 'Only a global domain is set' do
+      let(:config) { YAML.load(<<~EOT
+        ---
+        :config:
+          domain: example.com
+        :providers:
+          :mock_provider:
+        :pools:
+          - name: '#{pool}'
+            size: 1
+            provider: #{provider_name}
+        EOT
+      )}
+
+      it 'should return the domain set in the config section' do
+        allow(subject).to receive(:get_provider_for_pool).with(pool).and_return(provider)
+        allow(provider).to receive(:name).and_return(provider_name)
+        result = subject.get_domain_for_pool(pool)
+        expect(result).to_not be_nil
+        expect(result).to eq('example.com')
+      end
+    end
+
+    context 'A provider specified a domain to use' do
+      let(:config) { YAML.load(<<~EOT
+        ---
+        :config:
+        :providers:
+          :mock_provider:
+            domain: m.example.com
+        :pools:
+          - name: '#{pool}'
+            size: 1
+            provider: #{provider_name}
```

Comment:
another case could be added where `provider` is not set

Reply:
I just added such a test a little above here and am about po push the code.